docs/swagger.json

```diff
@@ -0,0 +1,549 @@
+                        "schema": {
+                            "$ref": "#/definitions/post_certificate_authority"
+                        }
+                    }
+                ],
+                "tags": [
+                    "/ca"
+                ]
+            },
+            "get": {
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/certificate_authorities_data"
+                        }
+                    }
+                },
+                "summary": "List all Certificate Authorities",
+                "operationId": "get_certificate_authority",
+                "tags": [
+                    "/ca"
+                ]
+            }
+        },
+        "/api/v1/ca/{ca_common_name}": {
+            "parameters": [
+                {
+                    "description": "Certificate Authority Common Name",
+                    "name": "ca_common_name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string"
+                }
+            ],
+            "get": {
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/get_common_name_response"
+                        }
+                    }
+                },
+                "summary": "Get information of specific Certificate Authority",
+                "operationId": "get_certificate_authority_management",
+                "tags": [
+                    "/ca"
+                ]
+            },
+            "put": {
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/get_common_name_response"
+                        }
+                    }
+                },
+                "summary": "Add Signed Certificate for Intermediate Certificate Authority",
+                "operationId": "put_certificate_authority_management",
+                "parameters": [
+                    {
+                        "name": "payload",
+                        "required": true,
+                        "in": "body",
+                        "schema": {
+                            "$ref": "#/definitions/certificate_intermediate_ca"
+                        }
+                    }
+                ],
+                "tags": [
+                    "/ca"
+                ]
+            }
+        },
+        "/api/v1/ca/{ca_common_name}/certificates": {
+            "parameters": [
+                {
+                    "name": "ca_common_name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string"
+                }
```

Comment:
Looking at the [rendered documentation](http://kairo.eti.br/ownca-rest-api/#/%2Fca/get_certificate_authority_management) it does show all the fields in the example response, but it should also show them in the parameters if they can be specified in the request.

Reply:
> Shouldn't we also add optional parameters to specify other parts of the certificate (oids, dns_names, etc...)?

Required by default is False. Do you mean ben explicitly with the ‘’required=False’’?

Reply:
> Looking at the [rendered documentation](http://kairo.eti.br/ownca-rest-api/#/%2Fca/get_certificate_authority_management) it does show all the fields in the example response, but it should also show them in the parameters if they can be specified in the request.

I don't follow. Can you explain? The request is made as a payload (JSON) in the body. Sorry, maybe I missed some detail. :)

Reply:
That was a good catch! Thanks!
I also fixed the response body.

What do you think about the ``required=False``, do we need to add it explicitly 🤔

Reply:
I think it doesn't affect readability much either way, so we could just add it explicitly in case someone wants it for some reason while ingesting the spec.

Reply:
Well, I don't think it would be a big deal either way. Your call

Reply:
That is a change that, if needed, we can work on further.